### PR TITLE
Add route info migration guide

### PR DESCRIPTION
**Pull request by chunhtai**
Migration guide for https://github.com/flutter/flutter/pull/119968

## Presubmit checklist
- [ ] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [ ] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [ ] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/release/breaking-changes/route-information-uri.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+description: Deprecation of `RouteInformation.location` and its related APIs
+---
+
+## Summary
+
+The APIs around the usages of RouteInformation.location was deprecated in the favor of RouteInformation.uri.
+
+## Context
+
+To handle mobile deeplinks from different web domains, the [`RouteInformation`][] needs to contain the authority information where the deeplinks are originated. The `uri` field was added to `RouteInformation` that captures the entire deeplink informations. Since then, every route-related parameters passing through framework were converted to full [`Uri`][] format. This led to deprecation of  imcompatible APIs.
```
Comment (loic-sharma) [inline src/release/breaking-changes/route-information-uri.md:12]: Consider something like:

```suggestion
The [`RouteInformation`][] needs the authority information to handle mobile deeplinks from different web domains. The `uri` field was added to `RouteInformation` that captures the entire deeplink information and route-related parameters were converted to the full [`Uri`][] format. This led to deprecation of incompatible APIs.
```

File: src/release/breaking-changes/route-information-uri.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
+---
+title: Migration guide for `RouteInformation.location`
+description: Deprecation of `RouteInformation.location` and its related APIs
+---
+
+## Summary
+
+The APIs around the usages of RouteInformation.location was deprecated in the favor of RouteInformation.uri.
```
Comment (loic-sharma) [inline src/release/breaking-changes/route-information-uri.md:8]: Consider something like:

```suggestion
`RouteInformation.location` and related APIs were deprecated in the favor of `RouteInformation.uri`.
```

File: src/release/breaking-changes/route-information-uri.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+```
+
+Code after migration:
+
+```dart
+SystemNavigator.routeInformationUpdated(uri: Uri.parse('/myLocation'));
+```
+
+
+## Timeline
```
Comment (loic-sharma) [inline src/release/breaking-changes/route-information-uri.md:85]:
````suggestion
```

## Timeline
````

Please ignore if this was intentional :)

**Review by sfshaza2 — APPROVED** [review]
lgtm

**Pull request merged**